client/response.go

```diff
@@ -73,11 +73,17 @@ func (r *Response) String() string {
 
 // JSON method will unmarshal body to json.
 func (r *Response) JSON(v any) error {
+	if r.client == nil {
```

Comment:
When will this be the case?

Reply:
Missing safety check found by https://github.com/uber-go/nilaway

Reply:
Are you sure it's not a false positive? When is `(*Response).JSON()` called without a `client`?
`nilaway` isn't really yet ready for productive use due to the many false-positives.

Reply:
Will remove